pupil-api/src/services/redis-pupil-auth.service.spec.ts

```diff
@@ -79,6 +85,46 @@ describe('redis-pupil-auth.service', () => {
     expect(payload).toEqual(expectedPayload)
   })
 
+  test('the check payload should not be returned if check pinValidFromUtc timestamp is after current date', async () => {
+    const currentDateTime = moment.utc()
+    const pinValidFromUtc = moment().add(1, 'hour')
+    const pinExpiresAtUtc = moment().add(2, 'hour')
+    const expectedPayload = {
+      config: {
+        practice: false
+      },
+      pinValidFromUtc: pinValidFromUtc,
+      pinExpiresAtUtc: pinExpiresAtUtc
+    }
+    redisServiceMock.get = jest.fn(async (key: string) => {
+      return expectedPayload
+    })
+    const schoolPin = 'abc12def'
+    const pupilPin = '5678'
+    const payload = await sut.authenticate(schoolPin, pupilPin)
+    expect(payload).toBeUndefined()
+  })
+
+  test('the check payload should not be returned if check pinExpiresAtUtc timestamp is before current date', async () => {
```

Comment:
Could you rephrase this test? Perhaps 'authorisation denied if the pin has expired' ?